src/content_script/input_tools.js

```diff
@@ -0,0 +1,237 @@
+      (element.type === "text" || element.type === "email")
+    );
+  },
+
+  addSLButtonToInput(inputElem, isInputFixed) {
+    // create wrapper for SL button
+    const btnWrapper = InputTools.newDiv("sl-button-wrapper");
+    const inputSumHeight = inputElem.getBoundingClientRect().height + "px";
+    btnWrapper.style.height = inputSumHeight;
+    btnWrapper.style.width = inputSumHeight;
+    document.body.appendChild(btnWrapper);
+
+    // create the SL button
+    const slButton = InputTools.newDiv("sl-button");
+    slButton.addEventListener("click", function () {
+      InputTools.handleOnClickSLButton(inputElem, slButton);
+    });
+    slButton.style.height = inputSumHeight;
+    slButton.style.width = inputSumHeight;
+    btnWrapper.appendChild(slButton);
+
+    InputTools.placeElementToTheRight(inputElem, btnWrapper, isInputFixed);
+  },
+
+  newDiv(classes) {
+    const div = document.createElement("div");
+    div.classList.add(classes);
+    return div;
+  },
+
+  placeElementToTheRight(anchor, elem, isFixed) {
+    let intervalId = 0;
+
+    function updatePosition() {
+      // check is element is removed from trackedElements
+      const i = InputTools.trackedElements.indexOf(anchor);
+      if (i === -1) {
+        elem.parentNode.removeChild(elem);
+        clearInterval(intervalId);
+      }
+
+      // get dimension & position of anchor
+      const anchorCoords = anchor.getBoundingClientRect();
+      const anchorStyle = getComputedStyle(anchor);
+      const elemWidth = InputTools.dimensionToInt(elem.style.width);
+      const pageXOffset = isFixed ? 0 : window.pageXOffset;
+      const pageYOffset = isFixed ? 0 : window.pageYOffset;
+      const buttonXOffset =
+        SLSettings.SLButtonPosition === "right-inside"
+          ? -elemWidth * 1.1
+          : elemWidth * 0.1;
+
+      // calculate elem position
+      const left =
+        anchorCoords.left +
+        pageXOffset +
+        anchor.offsetWidth +
+        buttonXOffset -
+        InputTools.dimensionToInt(anchorStyle.paddingRight) +
+        "px";
+
+      const top = anchorCoords.top + pageYOffset + "px";
+
+      if (isFixed) {
+        elem.style.position = "fixed";
+      }
+
+      if (elem.style.left !== left) {
+        elem.style.left = left;
+      }
+
+      if (elem.style.top !== top) {
+        elem.style.top = top;
+      }
+    }
+
+    intervalId = setInterval(updatePosition, 200);
+    updatePosition();
+  },
+
+  async handleOnClickSLButton(inputElem, slButton) {
+    if (InputTools.isLoading) {
+      return;
+    }
+    InputTools.isLoading = true;
+    slButton.classList.add("loading");
+
+    let res = await sendMessageToBackground("NEW_RANDOM_ALIAS");
+    if (res.error) {
+      alert("SimpleLogin Error: " + res.error);
+      res = { alias: "" };
+    }
+
+    InputTools.isLoading = false;
+    slButton.classList.remove("loading");
+
+    inputElem.value = res.alias;
+  },
+
+  sumPixel(dimensions) {
```

Comment:
this function doesn't seem to be used anywhere

Reply:
It is used in an older version. For now, maybe we can modify so it can be used in `placeElementToTheRight => calculate elem position` ?

Reply:
a3b19ef0dddcbcc92c906b29f939a24d2f3183a8